fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -359,6 +379,21 @@ public static Map<String, Set<String>> getMVPartitionNameMapOfExternalTable(Tabl
         return mvPartitionKeySetMap;
     }
 
+    // Iceberg Table has partition transforms like this:
+    // partition column is (ts timestamp), table could partition by year(ts), month(ts), day(ts), hour(ts)
+    // this function could get the dateInterval from partition transform
+    public static DateInterval getDateInterval(Table table, Column partitionColumn) {
+        if (partitionColumn.getType() != Type.DATE && partitionColumn.getType() != Type.DATETIME) {
+            return DateInterval.NONE;
+        }
+        if (table.isIcebergTable()) {
```

Comment:
Why depends on the table type, but not the partition type ?

Reply:
only iceberg table need add one year/month/hour, I do not know how to get the interval from partition type